refinery/data_set_manager/views.py

```diff
@@ -1170,11 +1182,97 @@ def patch(self, request, uuid):
                 return Response(
                     NodeSerializer(node).data, status=status.HTTP_200_OK
                 )
-            return Response('Currently, you can only remove node files.',
+            # derived node can have multiple attribute sources, some with
+            # the same values
+            elif solr_name and attribute_value and not node.is_derived():
+                # splits solr name into type and subtype
+                attribute_obj = customize_attribute_response([solr_name])[0]
+                root_node, root_attributes = self.find_root_attributes(
+                    [node], attribute_obj
+                )
+                # avoid issues with duplicate attributes for a node
+                for attribute in root_attributes:
+                    attribute.value = attribute_value
+                    attribute.save()
+                # start with source/start nodes and traverse down path
+                # over request file node and all other children
+                nodes_to_check = list(root_node.children.all())
+                updated_nodes = [root_node]
+                while nodes_to_check:
+                    current_node = nodes_to_check.pop()
+                    if current_node not in updated_nodes:
+                        self.update_annotated_nodes(current_node,
+                                                    attribute_obj,
+                                                    attribute_value)
+                        updated_nodes.append(current_node)
+                    children_nodes = current_node.children.all()
+                    for child_node in children_nodes:
+                        if child_node not in nodes_to_check:
+                            nodes_to_check.append(child_node)
+
+                return Response(
+                    NodeSerializer(node).data, status=status.HTTP_200_OK
+                )
+
+            return Response('Currently, you can only remove node files or '
+                            'edit non-derived files',
                             status=status.HTTP_405_METHOD_NOT_ALLOWED)
 
         return Response(uuid, status=status.HTTP_401_UNAUTHORIZED)
 
+    def find_root_attributes(self, parent_nodes, attribute_types,
```

Comment:
I think that this would still be more appropriate as a method on the `Node` model.

Consider its usage above: 
```
root_node, root_attributes = self.find_root_attributes(
     [node], attribute_obj
)
```

We already are operating on a single `Node` here, the wrapping in a list was a code smell to me.

I think the signature could be changed to: 
`def find_root_attributes(self, attribute_types, parent_nodes=None, checked_nodes=None):`

And the inner logic to:
```
        if checked_nodes is None:
            checked_nodes = []
        if parent_nodes is None:
            parent_nodes = self.parents.all()

        checked_nodes += [
            node for node in parent_nodes
            if node not in checked_nodes
        ]

        for parent_node in parent_nodes:
            node_attributes = Attribute.objects.filter(
                node=parent_node,
                type=attribute_types.get('attribute_type'),
                subtype=attribute_types.get('display_name').lower()
            )
            if node_attributes.exists():
                return parent_node, node_attributes
            else:
                return self.find_root_attributes(
                    attribute_types,
                    parent_nodes=parent_node.parents.all(),
                    checked_nodes=checked_nodes
                )
```